src/RA_Defs.h

```diff
@@ -19,9 +25,32 @@
 
 #else
 
+
+// Maybe an extra check just in-case
+
+#define _NORETURN            [[noreturn]]
+
+#if _HAS_CXX17
+#define _DEPRECATED          [[deprecated]]
+#define _DEPRECATEDR(reason) [[deprecated(reason)]]
+#define _FALLTHROUGH         [[fallthrough]]//; you need ';' at the end
+#define _UNUSED              [[maybe_unused]]
```

Comment:
_NODISCARD is already defined somewhere else in the STL and [[noreturn]] doesn't require C++17